### ISSUE TITLE
Cannot delete backups after upgrade to 2023.3.2

### ISSUE THREAD

**Issue by The00Dustin** [home-assistant/frontend#15787]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

When I hit the `DELETE` button under `Are you sure you want to delete this backup?` the backup is not deleted.

### Describe the behavior you expected

The backup should go away when I do this.

### Steps to reproduce the issue

1. Settings
2. System
3. Backups
4. Create Backup (Full)
5. Select Backup
6. Choose `Delete backup` option from kebab menu
7. Confirm deletion
8. See you are still in selected backup and it is not deleted

### What version of Home Assistant Core has the issue?

Not Shown...  See Additional Info

### In which browser are you experiencing the issue with?

iOS App, Safari, FireFox ESR

### Which operating system are you using to run this browser?

macSO Monterrey 12.6.3, iOS 16.3.1

### Additional information

Home Assistant 2023.3.2
Supervisor 2023.03.1
Operating System 9.5
Frontend 20230306.0 - latest

Prior to upgrading from 2023.3.1, I manually created a backup and deleted it successfully as I do before every upgrade (I create it, download it, and delete ahead of the upgrade instead of creating it during the upgrade because I'm temporarily running HAOS on an RPi microSD).

**Comment by The00Dustin** [comment]
Apparently I had a microSD card issue when creating the backup, a subsequent problem lead me to the error logs where I see I have a read-only filesystem.

**Comment by The00Dustin** [comment]
It turns out the microSD card issue occurred later.  After recovering from that, this backup (and some changes made after it) still existed and still wouldn't delete using the listed steps.  I created another backup and it also wouldn't delete.  I was able to delete both backups using these additional workaround steps:

9. Close the backup popup.
10. Select the backups using the checkboxes
11. Click `DELETE SELECTED BACKUPS` (on web interface) or tap the trash can icon (on iOS app)

After that, 2023.3.3 came out, so I created another backup (which I couldn't delete using the listed steps) and then upgraded (without deleting using the workaround).  After the upgrade, I still couldn't delete using the method I always used prior to 2023.3.2 (steps 1-8), but I could still delete using the workaround (steps 9-11).  I believe this consistent behavior most likely confirms this is a recently introduced bug vs a problem with my system.
